tests/test_cloudshell.py

```diff
@@ -0,0 +1,19 @@
+import unittest
+from msal.cloudshell import _scope_to_resource
+
+class TestScopeToResource(unittest.TestCase):
+
+    def test_expected_behaviors(self):
+        for scope, expected_resource in {
+            "https://analysis.windows.net/powerbi/api/foo":
+                "https://analysis.windows.net/powerbi/api",  # A special case
+            "https://pas.windows.net/CheckMyAccess/Linux/.default":
+                "https://pas.windows.net/CheckMyAccess/Linux/.default",  # Special case
+            "https://double-slash.com//scope": "https://double-slash.com/",
+            "https://single-slash.com/scope": "https://single-slash.com",
+            "guid/some/scope": "guid",
```

Comment:
Maybe using a real GUID to make it more reliable?

For example, add `0109ee35-235e-45dd-9fe4-371f53c66b2b/.default` -> `0109ee35-235e-45dd-9fe4-371f53c66b2b`.